sumpy/e2e.py

```diff
@@ -756,68 +743,36 @@ class M2LPostprocessLocal(E2EBase):
             m2l_translation.postprocess_local_nexprs(self.tgt_expansion,
-                                                     self.src_expansion)
-
-        tgt_coeff_exprs_before_postprocessing = [
-            sym.Symbol(f"tgt_coeff_before_postprocessing{i}")
-            for i in range(ncoeffs_before_postprocessing)]
-
-        src_rscale = sym.Symbol("src_rscale")
-        tgt_rscale = sym.Symbol("tgt_rscale")
-
-        from sumpy.assignment_collection import SymbolicAssignmentCollection
-        sac = SymbolicAssignmentCollection()
-
-        tgt_coeff_exprs = m2l_translation.postprocess_local_exprs(
-            self.tgt_expansion, self.src_expansion,
-            tgt_coeff_exprs_before_postprocessing,
-            sac=sac, src_rscale=src_rscale, tgt_rscale=tgt_rscale)
-
-        if result_dtype in (np.float32, np.float64):
-            real_func = sym.Function("real")
-            tgt_coeff_exprs = [real_func(expr) for expr in
-                    tgt_coeff_exprs]
-
-        tgt_coeff_names = [
-            sac.assign_unique(f"tgt_coeff{i}", coeff_i)
-            for i, coeff_i in enumerate(tgt_coeff_exprs)]
-
-        sac.run_global_cse()
+                self.src_expansion)
 
-        from sumpy.codegen import to_loopy_insns
-        return to_loopy_insns(
-                sac.assignments.items(),
-                vector_names={"d"},
-                pymbolic_expr_maps=[self.tgt_expansion.get_code_transformer()],
-                retain_names=tgt_coeff_names,
-                complex_dtype=to_complex_dtype(result_dtype),
-                )
+        child_knl = m2l_translation.postprocess_local_loopy_knl(
+            self.tgt_expansion, self.src_expansion, result_dtype)
 
-    def get_kernel(self, result_dtype):
-        ntgt_coeffs = len(self.tgt_expansion)
-        ntgt_coeffs_before_postprocessing = \
-            self.tgt_expansion.m2l_translation.postprocess_local_nexprs(
-                self.tgt_expansion, self.src_expansion)
         from sumpy.tools import gather_loopy_arguments
         loopy_knl = lp.make_kernel(
                 [
                     "{[itgt_box]: 0<=itgt_box<ntgt_boxes}",
-                    ],
+                    "{[isrc_coeff]: 0<=isrc_coeff<nsrc_coeffs}",
+                    "{[itgt_coeff]: 0<=itgt_coeff<ntgt_coeffs}",
+                ],
                 ["""
                 for itgt_box
-                """] + ["""
-                    <> tgt_coeff_before_postprocessing{idx} = \
-                            tgt_expansions_before_postprocessing[itgt_box, {idx}]
-                """.format(idx=i) for i in range(
-                    ntgt_coeffs_before_postprocessing)]
-                + self.get_loopy_insns(result_dtype) + ["""
-                    tgt_expansions[itgt_box, {idx}] = \
-                        tgt_coeff{idx}
-                    """.format(idx=i) for i in range(ntgt_coeffs)] + ["""
+                    <> coeffs[itgt_coeff] = 0 {id=init, dup=itgt_coeff}
```

Comment:
Use a better name for `coeffs`.

Reply:
Fixed in 25dd7fc4424695f26997e4bf2de55b7f1c0a25c2